src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -989,32 +1029,21 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
   * @define coll lazy list
   * @define Coll `LazyList`
   */
-@SerialVersionUID(3L)
+@SerialVersionUID(4L)
 object LazyList extends SeqFactory[LazyList] {
-  // Eagerly evaluate cached empty instance
-  private[this] val _empty = newLL(State.Empty).force
-
-  private sealed trait State[+A] extends Serializable {
-    def head: A
-    def tail: LazyList[A]
-  }
+  private object Uninitialized extends Serializable
+  private object MidEvaluation
 
-  private object State {
-    @SerialVersionUID(3L)
-    object Empty extends State[Nothing] {
-      def head: Nothing = throw new NoSuchElementException("head of empty lazy list")
-      def tail: LazyList[Nothing] = throw new UnsupportedOperationException("tail of empty lazy list")
-    }
-
-    @SerialVersionUID(3L)
-    final class Cons[A](val head: A, val tail: LazyList[A]) extends State[A]
-  }
+  private val EmptyFun = () => null
+  private val Empty = new LazyList(EmptyFun)
+  // Eagerly evaluate cached empty instance
+  locally(Empty.force)
```

Comment:
```suggestion
  private val Empty: LazyList[Nothing] = new LazyList(EmptyFun).force // Eagerly evaluate cached empty instance
```

Reply:
I should have added a comment, sorry. I did it separately because `force` calls `isEmpty`, which is `state eq LazyList.Empty`. The `state` method, after initialization, returns `LazyList.Empty`. But the field is not yet assigned, so `state` actually returns `null` and `isEmpty` ends up comparing `null eq null` - which is true, so it works out. But it feels off...